pkgs/top-level/all-packages.nix

```
@@ -25465,7 +25465,19 @@ with pkgs;
 
   ucommon = callPackage ../development/libraries/ucommon { };
 
-  v8 = darwin.apple_sdk_11_0.callPackage ../development/libraries/v8 { };
+  v8 = callPackage ../development/libraries/v8 (
+    let
+      stdenv' = if stdenv.cc.isClang && lib.versionAtLeast (lib.getVersion stdenv.cc.cc) "16"
+        then overrideCC stdenv (llvmPackages_15.clang.override {
+          inherit (stdenv.cc) libcxx;
+          extraPackages = [ stdenv.cc.libcxx.cxxabi ];
+        })
+        else stdenv;
```

I'm also needing this expression in places where `llvmPackages_*.stdenv` is used, as in #264830. Can we make something like `overrideLibcxx` in #265284 a stdenv adapter before #264091 is in, so that we can simply write the following?

```suggestion
      stdenv' = if stdenv.cc.isClang && lib.versionAtLeast (lib.getVersion stdenv.cc.cc) "16"
        then overrideLibcxx llvmPackages_15.stdenv
        else stdenv;
```

---

Sure, I can refactor it into an adapter.

---

There is one caveat. The libc++ 16 headers can’t be parsed by clang 14 or older. I’ll base the adapter’s implementation on #264091, so it can be used when a package needs an older version of clang to build. It can be deprecated once that PR is merged.